### PR TITLE
Port to dune (minimal version)

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Same as #59 but with as little change as possible to make it easier to review and merge

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
@ygrek CI is all green. This is ready to be merged.

File: test/runner.ml [hunk 1]
```diff
@@ -6,4 +6,18 @@ let () =
...
+    Test_DynArray.register;
+    Test_ExtArray.register;
+    Test_ExtHashtbl.register;
+    Test_ExtList.register;
+    Test_ExtString.register;
+    Test_IO.register;
+    Test_std.register;
+  ]
+  in
+  List.iter (fun register -> register ()) tests;
```
Comment (ygrek) [inline test/runner.ml:21]: why is this needed? dune will not link if no explicit dep?

**Review by ygrek — APPROVED** [review]

File: test/runner.ml (hunk 1 above)
Comment (kit-ty-kate) [reply test/runner.ml:21]: I’ve just tested and no. To be more explicit (and true), I’ve updated the comment in

https://github.com/ygrek/ocaml-extlib/pull/64/files#diff-6c3a27ff8273715c892c921fcc5edb627d8bd48baae795a8b2f8a463ccd431a7R27

to show this fact

**Comment by ygrek** [comment]
Thanks!
